### ISSUE TITLE
Add Fragment Example

### ISSUE THREAD

**Issue by mateusgrb** [mapbox/mapbox-android-demo#871]
I see only examples in activities. Fragment's lifecycle is a bit different so it'd be nice to have an example of a `MapView` in a fragment for guidance. Thank you!

**Comment by langsmith** [comment]
Awesome idea @mateusgrb . Thanks!

**Comment by mateusgrb** [comment]
I see `SupportMapFragment` is a built-in component from the library, I was thinking more of adding a `MapView` to my own fragment.

The use case is an activity that hosts a flow. Each step of the flow has its own fragment. At some point I switch from fragment A(which contains a MapView) to fragment B(which also contains its own  MapView). I already noticed, for example, that I can't call `mapView.onDestroy()` when moving from one fragment to another.

**Comment by mateusgrb** [comment]
@langsmith thanks for the tip, just added those.

**Comment by mateusgrb** [comment]
@langsmith do you also recommend calling `mapboxMap.clear()` in the fragment's `onDestroyView()`?

**Comment by langsmith** [comment]
You can if you'd like. I don't think it'll make any difference. `clear()` is about removing annotations and other visuals from the map:

```java
  /**
   * Removes all markers, polylines, polygons, overlays, etc from the map.
   */
  public void clear() {
    annotationManager.removeAnnotations();
  }
```

You still seeing issues @mateusgrb ?

**Comment by mateusgrb** [comment]
nope, not seeing any issues. Just bumped into that method and thought if it'd be a good practice to call it in `onDestroyView()`.

**Comment by langsmith** [comment]
Oh, ok 👍
